daprovider/reader.go

```diff
@@ -54,6 +60,13 @@ type Reader interface {
 		batchBlockHash common.Hash,
 		sequencerMsg []byte,
 	) containers.PromiseInterface[PreimagesResult]
+
+	// RecoverPayloadAndPreimages fetches the underlying payload and collects preimages from the DA provider given the batch header information
+	RecoverPayloadAndPreimages(
```

Comment:
Is there a strong reason we need to add this interface? Given that its usage is not wired up fully yet I'm not sure how it's going to be used.